hca/upload/api_client.py

```diff
@@ -1,81 +1,238 @@
-import json
-
 import requests
 from tenacity import retry, stop_after_attempt, wait_fixed
 
 from .upload_config import UploadConfig
 
 
+class UploadApiException(RuntimeError):
+    """
+    Exception returned if problems are encountered
+    """
+    pass
+
+
 class ApiClient:
```

Comment:
Should all the methods in this class be wrapped in retries?

Reply:
Ditto.